sklearn/gaussian_process/kernels.py

```diff
@@ -1141,14 +1153,23 @@ def __call__(self, X, Y=None, eval_gradient=False):
             hyperparameter of the kernel. Only returned when eval_gradient
             is True.
         """
+        length_scale = self.length_scale
+        if np.iterable(length_scale):
```

Comment:
Is this logic just `np.squeeze(length_scale).astype(np.float)`? Could we factor out `length_scale = _check_length_scale(X, length_scale)`?